### ISSUE TITLE
Rethink when to call BlueskyEventStream._load_partitions

### ISSUE THREAD

**Issue by gwbischof** [bluesky/databroker#509]
I think this may be an issue if the if load_partitions is first called before all of the documents are available. The next time you call read() on the stream, you may not get all of the documents, because load_partitions is not called a second time.

https://github.com/bluesky/databroker/blob/master/databroker/core.py#L1397

**Comment by danielballan** [comment]
Indeed, DataSource does not have the `reload()` concept that Catalog does. Perhaps it should.
